src/graphql-aspnet/Defaults/DefaultGraphQLHttpProcessor{TSchema}.cs

```diff
@@ -88,28 +97,112 @@ public class DefaultGraphQLHttpProcessor<TSchema> : IGraphQLHttpProcessor<TSchem
+                }
+            }
+            else if (string.Equals(context.Request.Method, nameof(HttpMethod.Get), StringComparison.OrdinalIgnoreCase))
+            {
+                queryData = this.DecodeAsGetRequest(context);
+            }
+            else
             {
                 await this.WriteStatusCodeResponse(HttpStatusCode.BadRequest, ERROR_USE_POST, context.RequestAborted).ConfigureAwait(false);
                 return;
             }
 
-            // accepting a parsed object causes havoc with any variables collection
+            await this.SubmitGraphQLQuery(queryData, context.RequestAborted).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Attempts to extract the necessary keys from the query string of the <paramref name="context"/>
+        /// and create a <see cref="GraphQueryData"/> object that can be processed by the GraphQL runtime.
+        /// </summary>
+        /// <param name="context">The http context to deserialize.</param>
+        /// <returns>GraphQueryData.</returns>
+        private GraphQueryData DecodeAsGetRequest(HttpContext context)
+        {
+            var queryString = string.Empty;
+            var variables = string.Empty;
+            var operationName = string.Empty;
+
+            if (context.Request.Query.ContainsKey(Constants.Web.QUERYSTRING_QUERY_KEY))
+                queryString = context.Request.Query[Constants.Web.QUERYSTRING_QUERY_KEY];
+
+            if (context.Request.Query.ContainsKey(Constants.Web.QUERYSTRING_OPERATIONNAME_KEY))
+                operationName = context.Request.Query[Constants.Web.QUERYSTRING_OPERATIONNAME_KEY];
+
+            if (context.Request.Query.ContainsKey(Constants.Web.QUERYSTRING_VARIABLES_KEY))
+                variables = context.Request.Query[Constants.Web.QUERYSTRING_VARIABLES_KEY];
+
+            return new GraphQueryData()
+            {
+                Query = queryString,
+                OperationName = operationName,
+                Variables = InputVariableCollection.FromJsonDocument(variables),
+            };
+        }
+
+        /// <summary>
+        /// Attempts to deserialize the POST body of the <paramref name="context"/>
+        /// into a <see cref="GraphQueryData"/> object that can be processed by the GraphQL runtime.
+        /// </summary>
+        /// <param name="context">The http context to deserialize.</param>
+        /// <returns>GraphQueryData.</returns>
+        protected async Task<GraphQueryData> DecodeAsPostRequest(HttpContext context)
+        {
+            // if a "query" parameter appears in the query string treat the request like a get
+            // request and parse parameters from the query string
+            // -----
+            // See: https://graphql.org/learn/serving-over-http/#http-methods-headers-and-body
+            // -----
+            if (context.Request.Query.ContainsKey(Constants.Web.QUERYSTRING_QUERY_KEY))
+                return this.DecodeAsGetRequest(context);
```

Comment:
I think the spec is a bit vague in:
> POST request [...] If the "query" query string parameter is present (as in the GET example above), it should be parsed and handled in the same way as the HTTP GET case.

Looking at the [reference implementation](https://github.com/graphql/express-graphql/blob/v0.12.0/src/index.ts#L492), they support mix and match, i.e. getting query from the url and variables from the body and even other combinations.

But in any case, I guess a more opinionated implementation like this one is ok too. i.e. everything on url or everything in body, no mix and match.

Reply:
I agree its a bit vague.  

I was on the fence and couldn't decide which should take priority in a scenario where both are present.   My initial thought was to adhere more closely to the verbiage of the spec rather than convention of other libraries, but there is value in preserving expectations. 

Rather than locking the library into one implementation or the other I will add support for "mix and match" approach by default and add configuration options to change the extraction behavior to be more rigid if necessary.

I appreciate the feedback!
Kevin